lib/jekyll/configuration.rb

```diff
@@ -385,5 +386,24 @@ def check_maruku(config)
           "`_config.yml` file."
       end
     end
+
+    # Private: Checks if the `plugins` config is a String
+    #
+    # config - the config hash
+    #
+    # Raises a Jekyll::Errors::InvalidConfigurationError if the config `plugins`
+    # is a string
+    private
+    def check_plugins(config)
+      if config.key?("plugins") && config["plugins"].is_a?(String)
+        Jekyll.logger.error "Configuration Error:", "You specified the" \
+          " `plugins` config in your configuration file as a string, please" \
+          " use an array instead. If you wanted to set the directory of your" \
+          " plugins, use the config key `plugins_dir` instead."
```

Comment:
`<<-TEXT` would be a deoptimization as you would have to strip the heredoc and multiple lines to get it to display nicely (so that it can fold.) This avoids that issue while making it readable (albeit more annoying for the developer.)